js-envs/src/main/scala/org/scalajs/jsenv/nodejs/NodeJSEnv.scala

```diff
@@ -79,244 +58,18 @@ class NodeJSEnv private (
-                // We must terminate with an error
-                process.exit(-1);
-              }
-            });
-          },
-          send: function(msg) {
-            if (socket === null) throw new Error("Com not open");
-
-            var len = msg.length;
-            var buf = new Buffer(4 + len * 2);
-            buf.writeInt32BE(len, 0);
-            for (var i = 0; i < len; ++i)
-              buf.writeUInt16BE(msg.charCodeAt(i), 4 + i * 2);
-            socket.write(buf);
-          },
-          close: function() {
-            if (socket === null) throw new Error("Com not open");
-            socket.end();
-          }
-        }
-      }).call(this);
-      """
-    )
-
-    def send(msg: String): Unit = {
-      if (awaitConnection()) {
-        jvm2js.writeInt(msg.length)
-        jvm2js.writeChars(msg)
-        jvm2js.flush()
-      }
-    }
-
-    def receive(timeout: Duration): String = {
-      if (!awaitConnection())
-        throw new ComJSEnv.ComClosedException("Node.js isn't connected")
-
-      js2jvm.mark(Int.MaxValue)
-      val savedSoTimeout = comSocket.getSoTimeout()
-      try {
-        val optDeadline = OptDeadline(timeout)
-
-        comSocket.setSoTimeout((optDeadline.millisLeft min Int.MaxValue).toInt)
-        val len = js2jvm.readInt()
-        val carr = Array.fill(len) {
-          comSocket.setSoTimeout((optDeadline.millisLeft min Int.MaxValue).toInt)
-          js2jvm.readChar()
-        }
-
-        js2jvm.mark(0)
-        String.valueOf(carr)
-      } catch {
-        case e: EOFException =>
-          throw new ComJSEnv.ComClosedException(e)
-        case e: SocketTimeoutException =>
-          js2jvm.reset()
-          throw new TimeoutException("Timeout expired")
-      } finally {
-        comSocket.setSoTimeout(savedSoTimeout)
-      }
-    }
-
-    def close(): Unit = {
-      serverSocket.close()
-      if (jvm2js != null)
-        jvm2js.close()
-      if (js2jvm != null)
-        js2jvm.close()
-      if (comSocket != null)
-        comSocket.close()
-    }
-
-    /** Waits until the JS VM has established a connection or terminates
-     *  @return true if the connection was established
-     */
-    private def awaitConnection(): Boolean = {
-      serverSocket.setSoTimeout(acceptTimeout)
-      while (comSocket == null && isRunning) {
-        try {
-          comSocket = serverSocket.accept()
-          jvm2js = new DataOutputStream(
-              new BufferedOutputStream(comSocket.getOutputStream()))
-          js2jvm = new DataInputStream(
-              new BufferedInputStream(comSocket.getInputStream()))
-        } catch {
-          case to: SocketTimeoutException =>
-        }
-      }
-
-      comSocket != null
-    }
-
+      extends AsyncNodeRunner(libs, code) with NodeComJSRunner {
     override protected def initFiles(): Seq[VirtualJSFile] =
-      super.initFiles :+ comSetup
-
-    override protected def finalize(): Unit = close()
+      super.initFiles ++ comSetup
```

Comment:
It seems both `ComDOMNodeRunner`and this class (`ComNodeRunner`) have the same implementation of `initFiles()`. Why is it not defined in `NodeComJSRunner`.